test/junit/scala/collection/mutable/HashSetTest.scala

```diff
@@ -70,4 +70,18 @@ class HashSetTest {
     val hs = HashSet[PackageEntryImpl](PackageEntryImpl("javax"), PackageEntryImpl("java"))
     assertFalse(hs.add(PackageEntryImpl("java")))
   }
+
+  class Type1(val value:String)
+  class Type2(val value:String)
+
+  import scala.jdk.CollectionConverters._
+
+  @Test
+  def addAll: Unit = {
+    val jhs: java.util.HashSet[Type1] = new java.util.HashSet[Type1]
+    jhs.add(new Type1("A"))
+    jhs.add(new Type1("B"))
+    val shs: scala.collection.mutable.Set[Type2] = jhs.asScala.map(x => new Type2(x.value))
+    assertTrue(jhs.size.equals(shs.size))
```

Comment:
assertEquals. Actually, I'm not sure what this test is testing, because there are a few moving parts. (It is not minimal.)

Reply:
It's a (slightly modified) copy of the minimised case in the issue.  Perhaps we can keep this as a unit test, but differentiate it by location and/or name?

Reply:
thanks, that explains what it has to do with the price of beans, as they say.